authority/provisioner/provisioner.go

```diff
@@ -351,13 +351,14 @@ func DefaultIdentityFunc(ctx context.Context, p Interface, email string) (*Ident
 		if !sshUserRegex.MatchString(name) {
 			return nil, errors.Errorf("invalid principal '%s' from email '%s'", name, email)
 		}
-		usernames := []string{name}
+		usernames := append(usernames, name)
```

Comment:
As it is you are always passing one username and the value will be empty for most IdPs, we should not send it in oidc.go or not append here if it's an empty string.

Reply:
As a variadic argument, wouldn't it default to zero length slice if no input is passed? Should it be checked there or at `oidc.go`?

Reply:
Yes I was talking about just a `""`, that it will be the default in most OIDC providers. The sanitation function fixed that.